File: catalogue-api/src/test/java/io/quartic/catalogue/api/PostgresDatasetLocatorShould.java
package io.quartic.catalogue.api;

public class PostgresDatasetLocatorShould extends DatasetLocatorTests<PostgresDatasetLocator> {
    @Override
    protected PostgresDatasetLocator locator() {
        return PostgresDatasetLocatorImpl.of("alice", "pass", "foo", "q");
    }

    @Override
    protected String json() {
        return "{\"type\": \"postgres\", \"user\": \"alice\", \"password\": \"pass\", \"url\": \"foo\", \"query\": \"q\"}";
    }
}

File: catalogue-api/src/test/kotlin/io/quartic/catalogue/api/PostgresDatasetLocatorShould.kt
package io.quartic.catalogue.api

class PostgresDatasetLocatorShould : DatasetLocatorTests<PostgresDatasetLocator>() {
    override fun locator() = PostgresDatasetLocator("alice", "pass", "foo", "q")

    override fun json() = "{\"type\": \"postgres\", \"user\": \"alice\", \"password\": \"pass\", \"url\": \"foo\", \"query\": \"q\"}"
}